tests/func/test_lock.py

```diff
@@ -28,6 +28,7 @@ def test_unlock_lock_failed(tmp_dir, dvc, mocker):
     with lock:
         lock.unlock()
         lock_ext.lock()  # imitate an exernal process had time to lock it
+        request.addfinalizer(lock_ext.unlock)
```

Comment:
`zc.Lockfile` keeps a file open on initialization that gets closed during `unlock()`.
So we should be explicitly closing this lockfile on teardown.